src/test/groovy/BusinessScenariosTests.groovy

```diff
@@ -0,0 +1,162 @@
+
+        createdContact = client.contacts().create(new Contact(name: testContactName, ownerId: userSalesRep.id, isOrganization: true))
+        await().until {
+            client
+                    .contacts()
+                    .list(new ContactsService.SearchCriteria().name(createdContact.name))
+                    .size() == 1
+        }
+    }
+
+    def convertUserId(String userId) {
+        assert userId
+        return new Long(userId)
+    }
+
+    def cleanupSpec() {
+        client
+                .deals()
+                .list(new DealsService.SearchCriteria().contactId((Long) createdContact.id))*.id
+                .each {
+            id ->
+                println(id)
+                client.deals().delete(id)
+        }
+
+        await().until {
+            client
+                    .deals()
+                    .list(new DealsService.SearchCriteria().contactId((Long) createdContact.id))
+                    .size() == 0
+        }
+
+        client
+                .contacts()
+                .delete(createdContact.id)
+
+
+        Contact tempContact = client
+                .contacts()
+                .list(new ContactsService.SearchCriteria().name((String) personContactName))[0]
+
+        if (null != tempContact) {
+            client
+                    .contacts()
+                    .delete(tempContact.id)
+        }
+    }
+
+    def getDealsStageCategory(long dealStageId, boolean active) {
+        Optional<Stage> result = client.stages()
+                .list(new StagesService.SearchCriteria().active(active))
+                .stream()
+                .filter { stage -> stage.getId().equals(dealStageId) }
+                .findAny();
+
+        assert result.present
+        result.get().category
+    }
+
+    def "new deal in incoming stage should be created automatically when new contact organization is assigned to a sales rep"() {
+        when:
+        await().sleep(40000)
+        Deal deal = client.deals().list(new DealsService.SearchCriteria().contactId((Long) createdContact.id))[0]
+        DateTimeFormatter dtfOut = DateTimeFormat.forPattern("dd/MM/yyyy")
+        String dealCreationDate = dtfOut.print(deal.createdAt)
+
+        then:
+        createdContact.isOrganization
+        deal.name
+        "${createdContact.name} ${dealCreationDate}" == deal.name
+        deal.contactId == createdContact.id
+        "incoming" == getDealsStageCategory(deal.stageId, true)
+        "active" == userSalesRep.status
+        userSalesRep.email.contains("_salesrep@")
+        !userSalesRep.email.contains("_accmanager@")
+    }
+
+    def "won deal should be assigned to account manager user and contact reassigned from sales rep to acc manager user"() {
+        when:
+        Deal createdDeal = client.deals().list(new DealsService.SearchCriteria().contactId(createdContact.id))[0]
+        Stage stage = client.stages().list(new StagesService.SearchCriteria().name("Won"))[0]
```

Comment:
- maybe we need to extract this line to method
- first stage name is "incoming"